File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 18 and 19 July 2020/04. Balls/04. Balls.py
balls_points = {'red': 5, 'orange': 10, 'yellow': 15, 'white': 20, 'black': 0}
balls_quantities = {'Total points': 0, 'red': 0, 'orange': 0, 'yellow': 0, 'white': 0, 'Other colors picked': 0,'Divides from black balls': 0}

for _ in range(int(input())):
    ball = input()

    if ball == 'black':
        balls_quantities['Divides from black balls'] += 1
        balls_quantities['Total points'] /= 2
        continue
    if ball in balls_points:
        balls_quantities[ball] += 1
        balls_quantities['Total points'] += balls_points[ball]
    else:
        balls_quantities['Other colors picked'] += 1

for k, v in balls_quantities.items():
    print(f'{k}: {int(v)}' if not k.find('T') else f'{k.title()} balls: {v}' if len(k) < 10 else f'{k}: {v}')


# ------------------------------------------------------(2)---------------------------------------------    
    
    
from math import floor

balls_count = int(input())
points = 0
balls_obj = {'red': 5, 'orange': 10, 'yellow': 15, 'white': 20, 'black': 0}
statistics_obj = {
    'red': 0,
    'orange': 0,
    'yellow': 0,
    'white': 0,
    'Other colors picked': 0,
    'Divides from black balls': 0}

for _ in range(balls_count):
    current_ball = input()
    if current_ball not in balls_obj:
        statistics_obj['Other colors picked'] += 1
        continue
    if current_ball == 'black':
        statistics_obj['Divides from black balls'] += 1
        points = floor(points / 2)
        continue
    statistics_obj[current_ball] += 1
    points += balls_obj[current_ball]


print('Total points:', points)

for k, v in statistics_obj.items():
    if len(k) < 10:
        print(f'Points from {k} balls: {v}')
    else:
        print(f'{k}: {v}')

'''
3
white
black
pink
'''

File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 18 and 19 July 2020/04. Balls/04. Balls.js
function balls(arg) {
    arg.shift()
    let pts = 0
    const ballsObj = { 'red': 5, 'orange': 10, 'yellow': 15, 'white': 20, 'black': () => pts /= 2 }

    const statisticsObj = {'red': 0,'orange': 0,'yellow': 0,'white': 0,'Other colors picked': 0,'Divides from black balls': 0}

    for (const ball of arg) {
        if (ballsObj[ball]) {
             if (ball == 'black') {
                ballsObj[ball]()
                statisticsObj['Divides from black balls'] += 1
             } else {
                 pts += ballsObj[ball]
                statisticsObj[ball] += 1
            }
        
        } else {
            statisticsObj['Other colors picked'] += 1
         }
            
    }

    console.log('Total points:', Math.floor(pts))
    for (const [key, val] of Object.entries(statisticsObj)) {
        console.log(key.length < 10 ? `Points from ${key} balls: ${val}` : `${key}: ${val}`)
    }
}

// balls(["3", "white", "black", "pink"])
// balls(["5", "red", "red", "ddd", "ddd", "sss"])